server/py/services/api/api/endpoints/model_monitoring.py

```diff
@@ -174,7 +174,8 @@ async def enable_model_monitoring(
     :param fetch_credentials_from_sys_config: Deprecated. If true, fetch the credentials from the system configuration.
 
     """
-    commons.get_monitoring_deployment().deploy_monitoring_functions(
+    await run_in_threadpool(
+        commons.get_monitoring_deployment().deploy_monitoring_functions,
```

Comment:
if there's no need for this function to be async - let it be sync and let fastapi handling thread allocation.